### ISSUE TITLE
pyright does not accept None as a valid input to a slice

### ISSUE THREAD

**Issue by FedericoV** [microsoft/pyright#154]
**Describe the bug**

If you want to slice a Python list to include the last element, the easiest way is to use None as an indexer, ie:

```
idx = None
a = [1, 2, 3, 4]
a[:idx]
```
Gives:
```
[1, 2, 3, 4]
```
Pyright complains that None is not a valid input into a slice.  IE, in visual studio code, the snippet gives the following pyright error:
```
Index for slice operation must be an integer value
```

**Comment by erictraut** [comment]
Thanks for the bug report. I didn't realize that None was a valid value, but I found in the spec where it is documented. This issue will be fixed in the next release of pyright.

**Comment by erictraut** [comment]
This is now fixed in version 1.0.41, which I just published. Thanks again for the bug report!
